backend/de.metas.business/src/main/java/de/metas/project/budget/BudgetProjectResourceService.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * #%L
+ * de.metas.business
+ * %%
+ * Copyright (C) 2022 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.project.budget;
+
+import de.metas.product.ResourceId;
+import de.metas.resource.Resource;
+import de.metas.resource.ResourceGroup;
+import de.metas.resource.ResourceGroupId;
+import de.metas.resource.ResourceService;
+import de.metas.resource.ResourceType;
+import de.metas.uom.IUOMDAO;
+import de.metas.uom.UOMPrecision;
+import de.metas.uom.UomId;
+import de.metas.util.Services;
+import lombok.NonNull;
+import org.adempiere.exceptions.AdempiereException;
+import org.compiere.model.I_C_Project_Resource_Budget;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.util.Optional;
+
+@Service
+public class BudgetProjectResourceService
+{
+	private final IUOMDAO uomDAO = Services.get(IUOMDAO.class);
+
+	private final ResourceService resourceService;
+
+	public BudgetProjectResourceService(@NonNull final ResourceService resourceService)
+	{
+		this.resourceService = resourceService;
+	}
+
+	@NonNull
+	public Optional<BigDecimal> computePlannedDuration(@NonNull final I_C_Project_Resource_Budget budgetResource)
```

Comment:
why not returning `Duration` instead of `BigDecimal`?

Reply:
NOTE: in DurationUtils you have a couple of helper methods to deal with time conversion, Duration and BigDecimal

Reply:
update: now I saw that it was refactored from some private method and moved here and it's convenient to return BigDecimal.
In this case i would suggest to write a small javadoc and document which is the UOM of returned BigDecimal.